File: dsa/Tree/height_of_a_binary_tree.cpp
#include <iostream>
#include <vector>
#include <stack>

using namespace std;

struct Node
{
    int val;
    Node *left, *right;
    Node(int val) : val(val), left(nullptr), right(nullptr) {}
};

class Solution
{
public:
    int maxDepth(Node *root)
    {
        if (root == nullptr)
            return 0;

        return 1 + max(maxDepth(root->left), maxDepth(root->right));
    }
};

int main()
{
    Node *root = new Node(1);
    root->left = new Node(2);
    root->right = new Node(3);
    root->left->left = new Node(4);
    root->left->right = new Node(5);
    root->left->right->left = new Node(8);
    root->right->left = new Node(6);
    root->right->right = new Node(7);
    root->right->right->left = new Node(9);
    root->right->right->right = new Node(10);

    Solution sol;
    cout << sol.maxDepth(root) << endl;
}

File: dsa/Tree/height_of_a_binary_tree.py
from queue import Queue

class Node:
    def __init__(self, val) -> None:
        self.val = val
        self.left = None
        self.right = None

class Solution:
    def maxDepth(self, root):
        if root == None:
            return 0
        
        q = Queue()
        q.put(root)

        level = 0

        while not q.empty():
            size = q.qsize()

            for _ in range(size):
                node = q.get()

                if node.left is not None:
                    q.put(node.left)
                if node.right is not None:
                    q.put(node.right)

            level += 1
        
        return level
    

if __name__ == "__main__":
    root = Node(1)
    root.left = Node(2)
    root.right = Node(3)
    root.left.left = Node(4)
    root.left.right = Node(5)
    root.left.right.right = Node(6)
    root.left.right.right.right = Node(7)

    sol = Solution()
    depth = sol.maxDepth(root)

    print(depth)
        